index.js

```diff
@@ -17,233 +15,235 @@ try {
   TMP = os.tmpdir()
 }
 
-function Storage (chunkLength, opts) {
-  const self = this
-  if (!(self instanceof Storage)) return new Storage(chunkLength, opts)
-  if (!opts) opts = {}
-
-  self.chunkLength = Number(chunkLength)
-  if (!self.chunkLength) throw new Error('First argument must be a chunk length')
-  self.name = opts.name || path.join('fs-chunk-store', randombytes(20).toString('hex'))
-  self.addUID = opts.addUID
-
-  if (opts.files) {
-    self.path = opts.path
-    if (!Array.isArray(opts.files)) {
-      throw new Error('`files` option must be an array')
-    }
-    self.files = opts.files.map(function (file, i, files) {
-      if (file.path == null) throw new Error('File is missing `path` property')
-      if (file.length == null) throw new Error('File is missing `length` property')
-      if (file.offset == null) {
-        if (i === 0) {
-          file.offset = 0
-        } else {
-          const prevFile = files[i - 1]
-          file.offset = prevFile.offset + prevFile.length
-        }
+export default class Storage {
+  constructor (chunkLength, opts) {
+    const self = this
+    if (!(self instanceof Storage)) return new Storage(chunkLength, opts)
```

Comment:
Given it's no longer a function, this can be removed

Reply:
@DiegoRBaquero fixed `self`, removed and replaced with this and anon functions